docs/rfc/rfc-1.adoc

```diff
@@ -633,13 +633,138 @@ As time passes and operators drop out of the system, a wallet becomes at risk
+transaction is the closing wallet's main UTXO, and we create an output for
+every multiple of `wallet_transfer_max` we have in BTC in the closing wallet.
+
+The destination wallets are chosen for their public key hash's modulus distance
+to the closing wallet's public key hash (both represented as a number). Ties are
+broken by the destination with the greater public key hash, and if we run out of
+viable wallets, it is okay to reuse them (though we prefer not to).
+
+For all the below examples, say that the public key hashes can range from 0 to
+99 (mod 100). Wallet#43 Means that the PKH of the Wallet is 43. Real public key
+hashes and their max mod values will be much higher in practice!
+
+Exceeding Transfer Max Example:
+
+- Wallet#90 with balance `250 BTC` needs to close.
+- `wallet_transfer_max = 100 BTC`
+- Live wallet options are Wallet#30, Wallet#80, Wallet#25, Wallet#15
+- Calculate distances:
+- Wallet#30 = min(90 - 30, 130 - 90) = min(60, 40) = 40
+- Wallet#80 = min(90 - 80, 180 - 90) = min(10, 90) = 10
+- Wallet#25 = min(90 - 25, 125 - 90) = min(65, 35) = 35
+- Wallet#15 = min(90 - 15, 115 - 90) = min(75, 25) = 25
+
+Deposit 100 BTC into Wallet#80, 100 BTC into Wallet#15, 50 BTC into Wallet#25
+
+Under Transfer Max Example:
+
+- Wallet#52 with balance `50 BTC` needs to close.
+- `wallet_transfer_max = 100 BTC`
+- Live wallet options are Wallet#13, Wallet#90, Wallet#59, Wallet#42
+- Calculate distances:
+- Wallet#13 = min(52 - 13, 113 - 52) = min(39, 61) = 39
+- Wallet#90 = min(90 - 52, 152 - 90) = min(38, 62) = 38
+- Wallet#59 = min(59 - 52, 152 - 59) = min(7, 93) = 7
+- Wallet#42 = min(52 - 42, 142 - 52) = min(10, 90) = 10
+
+Since Wallet#59 is the closest and Wallet#52's balance is below
+`wallet_transfer_max`, we transfer the whole balance to Wallet#59.
+
+Tiebreak Example:
+
+- Wallet#52 with a balance `13 BTC` needs to close.
+- `wallet_transfer_max = 100 BTC`
+- Live wallet options are Wallet#13, Wallet#60, Wallet#44, Wallet#75
+- Calculate distances:
+- Wallet#13 = min(52 - 13, 113 - 52) = min(39, 61) = 39
+- Wallet#60 = min(60 - 52, 152 - 60) = min(8, 92) = 8
+- Wallet#44 = min(52 - 44, 144 - 52) = min(8, 92) = 8
+- Wallet#75 = min(75 - 52, 152 - 75) = min(23, 77) = 23
+
+Since Wallet#60 is tied with Wallet#44 for the closest, we favor Wallet#60
+because 60>44. Since Wallet#52's balance is below `wallet_transfer_max`, we
+transfer the whole balance to Wallet#60.
+
+Wallet Reuse Example:
+
+- Wallet#35 with a balance `413 BTC` needs to close.
+- `wallet_transfer_max = 100 BTC`
+- Live wallet options are Wallet#20, Wallet#30
+- Calculate distances:
+- Wallet#20 = min(35 - 20, 120 - 35) = min(15, 85) = 15
+- Wallet#30 = min(35 - 30, 130 - 35) = min(5, 95) = 5
+
+Deposit 100 BTC into Wallet#30, 100 BTC into Wallet#20, 100 BTC into Wallet#30,
+100 BTC into Wallet#20, 13 BTC into Wallet#30 for a total of 213 BTC into Wallet#30 and 200 BTC into Wallet#20.
```

Comment:
Do we expect the operators to commit to which wallets they are moving the funds before they actually do it?

Reply:
Answer: no. This would not be possible. We will allow anyone to challenge their choice later.